Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift

```diff
@@ -574,6 +574,31 @@ extension SwiftPMBuildSystem: SKCore.BuildSystem {
       "--disable-index-store",
       "--target", target.targetID,
     ]
+    if self.toolsBuildParameters.configuration != self.destinationBuildParameters.configuration {
+      logger.fault(
+        """
+        Preparation is assuming that tools and destination are built using the same configuration, \
+        got tools: \(String(describing: self.toolsBuildParameters.configuration), privacy: .public), \
+        destination: \(String(describing: self.destinationBuildParameters.configuration), privacy: .public)
+        """
+      )
+    }
+    arguments += ["-c", self.destinationBuildParameters.configuration.rawValue]
+    if self.toolsBuildParameters.flags != self.destinationBuildParameters.flags {
+      logger.fault(
+        """
+        Preparation is assuming that tools and destination are built using the same build flags, \
+        got tools: \(String(describing: self.toolsBuildParameters.flags)), \
+        destination: \(String(describing: self.destinationBuildParameters.configuration))
+        """
+      )
+      logger.fault("Preparation is assuming that tools and destination to have the same build flags")
```

Comment:
Duplicate, but also this is going to be fairly spammy.

Reply:
I don’t expect this ever to be logged because we set the same build flags for the tools and destination build parameters here: https://github.com/apple/sourcekit-lsp/blob/b9c74387f297147d82716195974306384ca59c11/Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift#L245-L259

The intention of the log is that if that assumption is broken in the future, we’ll have a log message that’s pointing us to it.